aiotfm/inventory.py

```diff
@@ -139,145 +147,184 @@ def cmp(a, b):
+	trader: `str`
+		The player the client is trading with.
+	locked: List[`bool`]
+		A list of two `bool` describing the locked state of each party.
+	imports: :class:`aiotfm.inventory.TradeContainer`
+		The container of the items you will receive if the trade succeed.
+	exports: :class:`aiotfm.inventory.TradeContainer`
+		The container of the items you will give if the trade succeed.
+	state: :class:`aiotfm.utils.TradeState`
+		The current state of the trade.
+			ON_INVITE: an invitation has been received from/sent to the other party.
+			ACCEPTING: the client accepted and is waiting for the other party to be ready.
+			TRADING:   the only state of the trade where you are able to add items.
+			CANCELLED: the trade has been cancelled by one of the parties.
+			SUCCESS:   the trade finished successfully."""
+	def __init__(self, client, trader):
+		self.client = client
+		self.trader = trader
+		self.locked = [False, False] # 0: client, 1: trader
+
+		self.imports = TradeContainer(self)
+		self.exports = TradeContainer(self)
+
+		self.state = TradeState.ON_INVITE
+		self.pid = -1
+
+		if isinstance(trader, Player):
+			if self.trader.isGuest:
+				raise TypeError("You can not trade with a guest.")
+			if self.trader == self.client.username:
+				raise TypeError("You can not trade with yourself.")
+			if self.trader.pid == 0:
+				raise TypeError("You can not trade with a player having the same IP.")
+			self.trader = self.trader.username
+			self.pid = trader.pid
+		elif isinstance(trader, str):
+			if '#' not in trader:
+				raise TypeError("The player tag is needed to begin a trade.")
+			if trader.startswith('*'):
+				raise TypeError("You can not trade with a guest.")
+		else:
+			raise TypeError(f"Trade excepted 'Player' or 'str' type, got '{type(trader)}")
 
 	def __repr__(self):
-		return "<Trade on_invite={} accepted={} alive={} canceled={} locked_me={} locked_other={} traders={}>".format(
-			self.on_invite, self.accepted, self.alive, self.canceled, self.locked_me, self.locked_other, self.traders
-		)
-
-	def _update_player(self, player):
-		self.traders[self._other_index] = self._other = player
-		if player.trade != self:
-			player.trade = self
-
-	def _close(self):
-		self.alive = False
-		if self._client.trade == self:
-			self._client.trade = None
-		if self._other.trade == self:
-			self._other.trade = None
-		if self in self._client.trades:
-			self._client.trades.remove(self)
+		return "<Trade state={} locked=[client:{}, trader:{}] trader={} pid={}>".format(TradeState[self.state], *self.locked, self.trader, self.pid)
+
+	def __eq__(self, other):
+		if self.pid == -1 or other.pid == -1:
+			return self.trader.lower() == other.trader.lower()
+		return self.pid == other.pid
+
+	@property
+	def closed(self):
+		"""Returns True if the trade is closed."""
+		return self.state in (TradeState.SUCCESS, TradeState.CANCELLED)
+
+	def _start(self, pid):
+		self.state = TradeState.TRADING
+		self.client.trade = self

```

Comment:
I guess this line will fit better in between the lines 258 and 259 of aiotfm/client.py.